packages/react-button/src/components/Button/Button.tsx

```diff
@@ -12,8 +11,20 @@ import { renderButton } from './renderButton';
 export const Button = React.forwardRef<HTMLElement, ButtonProps>((props, ref) => {
   const state = useButton(props, ref);
 
-  useButtonClasses(state);
-  useInlineTokens(state, '--button');
+  const receivedChildren = !!state.children?.children;
+  const receivedIcon = !!state.icon?.children;
+
+  const iconOnly = receivedIcon && !receivedChildren;
+  const styleSelectors: ButtonStyleSelectors = {
+    disabled: state.disabled,
+    primary: state.primary,
+    iconOnly: iconOnly,
+    textOnly: receivedChildren && !receivedIcon,
+    textWithIcon: receivedIcon && receivedChildren,
```

Comment:
It seems like there's currently no need for these two right now looking at `useButtonClasses` so far.

Reply:
Nice catch, removed. This makes me think we should have a safeguard against this somehow... I suppose the RFC on iterating styles will remove this need since we won't have matchers any longer.